projects/kit/components/tree/components/tree/tree.component.ts

```diff
@@ -48,10 +40,12 @@ export class TuiTreeComponent<T> implements DoCheck {
     @ViewChild(forwardRef(() => TuiTreeComponent))
     protected readonly child?: TuiTreeComponent<T>;
 
-    protected readonly children$ = this.check$.pipe(
-        startWith(null),
-        map(() => this.handler(this.value)),
-        distinctUntilChanged(),
+    protected readonly children = toSignal(
```

Comment:
`toSignal` immediately subscribes and at that time there's no `value`. I have a feeling this changes probably breaks something.

Reply:
Don't worry, be happy :)

Reply:
Unless we can be sure, I don't think we should merge that.